Shared/Samples/List geodatabase versions/ListGeodatabaseVersionsView.swift

```
@@ -0,0 +1,133 @@
+    )
+    /// The geodatabase version information.
+    @State private var geodatabaseVersionInfos: [GeodatabaseVersionInfo]?
+    
+    var body: some View {
+        Form {
+            if let geodatabaseVersionInfos {
+                ForEach(geodatabaseVersionInfos, id: \.objectID) { version in
+                    Section("OBJECTID: \(version.objectID)") {
+                        LabeledContent("Name", value: version.name)
+                            .lineLimit(1)
+                            .textSelection(.enabled)
+                        LabeledContent("Access", value: version.access)
+                        LabeledContent("Created", value: version.created.formatted(date: .abbreviated, time: .shortened))
+                        LabeledContent("Last Modified", value: version.lastModified.formatted(date: .abbreviated, time: .shortened))
+                        LabeledContent("Is Owner", value: version.isOwner)
+                        LabeledContent("Parent Version", value: version.parentVersionName)
+                        if !version.description.isEmpty {
+                            LabeledContent("Description", value: version.description)
+                        }
+                    }
+                }
+            } else {
+                ProgressView()
+                    .frame(maxWidth: .infinity)
+            }
+        }
+        .task {
+            do {
+                if let resultFeatures = try await fetchGeodatabaseVersions() {
+                    geodatabaseVersionInfos = makeGeodatabaseVersionInfos(from: resultFeatures)
+                }
+            } catch {
+                analysisError = error
+            }
+        }
+        .errorAlert(presentingError: $analysisError)
+    }
+    
+    /// Fetches geodatabase versions using a geoprocessing job.
+    private func fetchGeodatabaseVersions() async throws -> FeatureSet? {
```

Throwing and also returning an optional might not always be the best choice. Should you throw some well-defined error instead of returning `nil` or does an optional really make sense here?

---

From core yaml:

> (GeoprocessingFeatures.features) It can be nil for services that have a map service, in which case calling `fetchOutputFeatures()` will retrieve the features from the service.

Thus, as long as the service provided supports versioning, the output `featureSet` won't be `nil` if the geoprocessing job successfully finishes with the results. I decided to `fatalError` in the case of `nil`.